modules/Emails/Email.php

```diff
@@ -4684,15 +4711,19 @@ private function sendOptInEmailToEmailAddressById($id)
             isset($sugar_config['email_enable_auto_send_opt_in'])
             && $sugar_config['email_enable_auto_send_opt_in']
         ) {
-            /** @var \EmailAddress $emailAddress */
+            /** @var EmailAddress $emailAddress */
             $emailAddresses = BeanFactory::getBean('EmailAddresses');
             $emailAddress = $emailAddresses->retrieve($id);
 
             if (
                 $emailAddress !== null
                 && $emailAddress->getConfirmedOptInState() != EmailAddress::COI_STAT_CONFIRMED_OPT_IN
-                && empty($emailAddress->confirm_opt_in_sent_date)) {
-                $this->sendOptInEmail($emailAddress);
+                && empty($emailAddress->confirm_opt_in_sent_date)
```

Comment:
Can we use null instead of empty?

Reply:
I am not sure: It is a bean field come from db:email_addresses table. This field could be null ad db-value but I don't know what is it after a retrive() method call?